packages/core/src/utils.ts

```diff
@@ -479,3 +483,179 @@ export function capitalize(word: string): string {
 export function isLowerCase(c: string): boolean {
   return c === c.toLowerCase();
 }
+
+/**
+ * Tries to find a code string for a given system within a given codeable concept.
+ * @param concept The codeable concept.
+ * @param system The system string.
+ * @returns The code if found; otherwise undefined.
+ */
+export function getCodeBySystem(concept: CodeableConcept, system: string): string | undefined {
+  return concept?.coding?.find((coding) => coding.system === system)?.code;
+}
+
+/**
+ * Sets a code for a given system within a given codeable concept.
+ * @param concept The codeable concept.
+ * @param system The system string.
+ * @param code The code value.
+ */
+export function setCodeBySystem(concept: CodeableConcept, system: string, code: string): void {
+  if (!concept.coding) {
+    concept.coding = [];
+  }
+  const coding = concept.coding?.find((coding) => coding.system === system);
+  if (coding) {
+    coding.code = code;
+  } else {
+    concept.coding?.push({ system, code });
+  }
+}
+
+/**
+ * Tries to find an observation interval for the given patient and value.
+ * @param definition The observation definition.
+ * @param patient The patient.
+ * @param value The observation value.
+ * @returns The observation interval if found; otherwise undefined.
+ */
+export function findObservationInterval(
+  definition: ObservationDefinition,
+  patient: Patient,
+  value: number
+): ObservationDefinitionQualifiedInterval | undefined {
+  return definition.qualifiedInterval?.find(
+    (interval) =>
+      observationIntervalMatchesPatient(interval, patient) &&
+      observationIntervalMatchesValue(interval, value, definition.quantitativeDetails?.decimalPrecision)
+  );
+}
+
+/**
+ * Returns true if the patient matches the observation interval.
+ * @param interval The observation interval.
+ * @param patient The patient.
+ * @returns True if the patient matches the observation interval.
+ */
+function observationIntervalMatchesPatient(
```

Comment:
Naming question - `observationInterval` vs. `qualifiedInterval`. On the one hand `observationInterval` seems to use more everyday language. On the other, `qualifiedInterval` matches the name of the property inside the observation definition